### PR TITLE
Code clean up.

### PR DESCRIPTION
**Pull request by nekonomicon**

### CODE REVIEW

File: cl_dll/hl/hl_weapons.cpp [hunk 1]
```diff
@@ -621,7 +621,7 @@ void HUD_InitClientWeapons( void )
 	// Pass through to engine
 	g_engfuncs.pfnPrecacheEvent		= gEngfuncs.pfnPrecacheEvent;
 	g_engfuncs.pfnRandomFloat		= gEngfuncs.pfnRandomFloat;
-	g_engfuncs.pfnRandomLong		= RandomLong;
+	g_engfuncs.pfnRandomLong		= gEngfuncs.pfnRandomLong;
```
Comment (a1batross) [inline cl_dll/hl/hl_weapons.cpp]: There is a different funtion prototypes. Isn't it?

File: cl_dll/hud_spectator.cpp [hunk 2]
```diff
@@ -874,6 +874,9 @@ bool CHudSpectator::ParseOverviewFile( )
 	
 	char *pfile  = NULL;
 
+	memset( filename, 0, 255 );
+	memset( levelname, 0, 255 );
+	memset( token, 0, 1024 );
```
Comment (a1batross) [inline cl_dll/hud_spectator.cpp]: It's a C++. Why not to use array initializers?

File: cl_dll/hl/hl_weapons.cpp (hunk 1 above)
Comment (nekonomicon) [inline cl_dll/hl/hl_weapons.cpp]: Nope. It seems Uncle Mike changed it.

File: cl_dll/hud_spectator.cpp (hunk 2 above)
Comment (nekonomicon) [inline cl_dll/hud_spectator.cpp]: Ok. I'm rework it.

**Pull request merged**